docs/pages/framework/testing-cheat-codes.md

```diff
@@ -0,0 +1,242 @@
+
+- Testing a contract's access control mechanisms, such as only allowing certain addresses to call specific functions.
+- Simulating a scenario where a contract is called by a different address.
+
+```
+use starknet::{testing, get_caller_address, contract_address_const};
+
+let user_one = contract_address_const::<'user1'>();
+testing::set_caller_address(user_one);
+assert(get_caller_address() == user_one, 'bad caller';
+```
+
+3. `set_contract_address`
+
+This cheat code helps you set the contract address to the provided value, allowing you to test contract deployment and interactions.
+
+It is important to note that any test function is considered a contract, which by default uses the `0` address. Using `set_contract_address` allows you to mock the current address of the testing function, making it useful to call other contract that may use `get_caller_address`.
+
+```
+use starknet::{testing, get_contract_address, contract_address_const};
+
+const HUB_ADDRESS: ContractAddress = contract_address_const::<'hub'>()
+
+testing::set_contract_address(HUB_ADDRESS);
+
+assert(get_contract_address() == HUB_ADDRESS, 'BAD ADDRESS');
+```
+
+4. `set_block_timestamp`
+
+This cheat code helps you set the block timestamp to the specified value, allowing you to test contract behavior at different points in time. You can apply this code when:
+
+- Testing a contract's behavior at a specific point in time, such as checking if a certain function is only callable during a certain time period.
+- Simulating a scenario where a contract is deployed at a different point in time.
+
+```
+use starknet::{testing, get_block_timestamp};
+
+testing::set_block_timestamp(123456);
+
+assert(get_block_timestamp == 123456, 'bad timestamp');
+
+```
+
+5. `set_version`
+
+This cheat code helps one set the version to the provided value, enabling you to test contract behavior with different versions. You can apply this when:
+
+- Testing a contract's behavior with different versions, such as checking if a certain function is only callable in a specific version.
+- Simulating a scenario where a contract is upgraded to a different version.
+
+```
+use starknet::{testing, get_version};
+
+testing::set_version('0.1.0');
+
+assert(get_version == '0.1.0', 'wrong version')
+
+```
+
+6. `set_account_contract_address`
+
+This cheat code helps you set the account contract address to the provided value, allowing you to test contract interactions with different account contracts. You can apply this when:
+
+- Testing a contract's interaction with a specific account contract, such as a wallet contract.
+- Simulating a scenario where a contract is called by a different account contract.
+
+```
+use starknet::{testing, get_account_contract_address, contract_address_const};
+
+const contract: ContractAddress = contract_address_const::<'contract'>();
+
+testing::set_account_contract_address(contract);
+
+assert(get_account_contract_address() == contract, 'wrong contract address');
```

Comment:
```suggestion
assert_eq!(get_tx_info().unbox().account_contract_address.into(), contract);
```
Using `assert_eq!` you don't always require to put a reason. Can be sometimes shorter and better. :)